x-pack/platform/plugins/shared/triggers_actions_ui/public/types.ts

```diff
@@ -119,6 +120,29 @@ type ResolvedRule = Omit<
   actions: RuleUiAction[];
 };
 
+type PrebuildField<T> = (props: T) => {
+  title: string;
+  description: NonNullable<React.ReactNode>;
+};
+
+export interface PrebuildFieldsMap {
```

Comment:
Note for the future. I like that we handle internally some of the common rule params. I would anticipate the need to extend this to support more params not related to queries or index patterns. Maybe in the future we can extend this to allow custom fields (free form).